### PR TITLE
kubemacpool: Remove namespace labels

### PR DESCRIPTION
**Pull request by qinqon**
**What this PR does / why we need it**:
The kubemacpool project was adding restricted label to test security context. This change remove labels altogether and configure them at CI to test it.

**Special notes for your reviewer**:
Since the CNAO namespace is shared between components is done there instead of kubemacpool that has it's own namespace.

**Release note**:

```release-note
Remove kubemacpool namespace labels.
```

### CODE REVIEW

File: data/kubemacpool/kubemacpool.yaml [hunk 1]
```diff
@@ -1,9 +1,6 @@
 apiVersion: v1
 kind: Namespace
 metadata:
-  labels:
-    control-plane: mac-controller-manager
```
Comment (oshoval) [inline data/kubemacpool/kubemacpool.yaml:5]: why `control-plane: mac-controller-manager` is removed as well ?
it was not part of the latest breaking change

**Comment by oshoval** [comment]
Worth to enclose the failure of HCO please on PR desc

File: data/kubemacpool/kubemacpool.yaml (hunk 1 above)
Comment (RamLavi) [reply data/kubemacpool/kubemacpool.yaml:5]: +1, it should stay. I think we already delete it for non-namespaced objects ([here](https://github.com/kubevirt/cluster-network-addons-operator/blob/main/pkg/controller/networkaddonsconfig/networkaddonsconfig_controller.go#L537)).

File: automation/check-patch.e2e-kubemacpool-functests.sh [hunk 2]
```diff
@@ -38,6 +38,17 @@ main() {
     source automation/check-patch.setup.sh
     cd ${TMP_PROJECT_PATH}
 
+    kubectl apply -f - <<EOF
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: cluster-network-addons
+  labels:
+    control-plane: mac-controller-manager
```
Comment (RamLavi) [inline automation/check-patch.e2e-kubemacpool-functests.sh:47]: I think adding/removing the `control-plane` label should not be part of this PR,

File: data/kubemacpool/kubemacpool.yaml (hunk 1 above)
Comment (oshoval) [reply data/kubemacpool/kubemacpool.yaml:5]: Btw some of the logic there was in order to maintain backward compatability
maybe the versions we support /deployed on customer site are new enough so we can remove this patch?
(more info here https://github.com/kubevirt/cluster-network-addons-operator/pull/1145)

Comment (qinqon) [reply data/kubemacpool/kubemacpool.yaml:5]: I would not depend on the controller and keep the manifest as we want them to be, so we don't need time for them to be fixed.

File: hack/components/bump-kubemacpool.sh [hunk 3]
```diff
@@ -116,14 +120,18 @@ EOF
...
-    cat <<EOF > config/cnao/cnao_mac-range_patch.json
+    cat <<EOF > config/cnao/cnao_mac-range_patch.yaml
 - op: replace
   path: /data/RANGE_START
   value: "{{ .RangeStart }}"
 - op: replace
   path: /data/RANGE_END
   value: "{{ .RangeEnd }}"
 EOF
+    cat <<EOF > config/cnao/cnao_remove-labels_patch.yaml
```
[the remark this answers is not in the document]
Comment (qinqon) [reply hack/components/bump-kubemacpool.sh:131]: I am going to remove all the labels since controller is removing the other one too.

File: data/kubemacpool/kubemacpool.yaml (hunk 1 above)
Comment (oshoval) [reply data/kubemacpool/kubemacpool.yaml:5]: The controller is removing them only for backward compatibility, for versions where CNAO label stuff that it didn't need to have and then those versions were upgraded to one that fixed the labeling.

You can see here a discussion that tells that we better remove it from the source itself - KMP, IIRC
https://github.com/kubevirt/cluster-network-addons-operator/pull/1145/files#r796372158

Comment (oshoval) [reply data/kubemacpool/kubemacpool.yaml:5]: need to rerun bump right ? since you changed the script

**Comment by oshoval** [comment]
https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_cluster-network-addons-operator/1566/pull-e2e-cnao-multus-dynamic-networks-functests/1668906222926958592
we have this here btw (don't think it is related to this PR IIRC, but maybe worth to double check if interesting)

```
W0614 09:24:17.766599   36399 warnings.go:70] would violate PodSecurity "restricted:latest": allowPrivilegeEscalation != false (container "samplepod" must set securityContext.allowPrivilegeEscalation=false), unrestricted capabilities (container "samplepod" must set securityContext.capabilities.drop=["ALL"]), runAsNonRoot != true (pod or container "samplepod" must set securityContext.runAsNonRoot=true), seccompProfile (pod or container "samplepod" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost")
```

**Review by RamLavi — APPROVED** [review]
/lgtm

I think that removing all the labels in the yaml patch may bite us in the future, but we can remove the `control-plane` label  from the NS on the KMP repo in parallel, then consider refining the yaml patch to remove exactly the labels we want.

**Comment by oshoval** [comment]
it still fails

**Comment by qinqon** [comment]
hiccup 
```
pull-e2e-cluster-network-addons-operator-kubemacpool-functest
```

> /lgtm
> 
> I think that removing all the labels in the yaml patch may bite us in the future, but we can remove the `control-plane` label from the NS on the KMP repo in parallel, then consider refining the yaml patch to remove exactly the labels we want.

Is very dangerous to label the shared namespace, it affects all the network components, we should remove the label, if we see kubemacpool depends on something, we will fix it.

**Comment by oshoval** [comment]
Sumup here for future possible AI, thanks

https://github.com/kubevirt/cluster-network-addons-operator/issues/1567

File: automation/components-functests.setup.sh [hunk 4]
```diff
@@ -32,6 +32,18 @@ export KUBECONFIG=$(cluster::kubeconfig)
 make cluster-operator-push
 make cluster-operator-install
 
+# Test kubemacpool with restricted
+if [ "$COMPONENT" == "kubemacpool" ]; then
+    cluster/kubectl.sh apply -f - <<EOF
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: cluster-network-addons
```
Comment (oshoval) [inline automation/components-functests.setup.sh:41]: nit: we don't need to parameterize the ns name ?
anyhow if needed can be done on follow PR
